### ISSUE TITLE
[Feature] Optional Uptime SLA for AKS cluster

### ISSUE THREAD

**Issue by jluk** [Azure/AKS#1603]
**What happened**:
Uptime SLA is an optional feature to enable a financially backed higher SLA for a cluster. Uptime SLA guarantees 99.95% availability of the Kubernetes API server endpoint for clusters that use Availability Zones and 99.9% of availability for clusters that don't use Availability Zones.

Documentation:
https://docs.microsoft.com/en-us/azure/aks/uptime-sla

**Comment by bhicks329** [comment]
Is there a view of when this will be in UK South?

**Comment by flamingboo** [comment]
@jluk Thanks for the link. One quick question though:
The document on Uptime SLA mentions the following:
"For clusters with egress lockdown, see limit egress traffic to open appropriate ports for Uptime SLA".
The link https://docs.microsoft.com/en-us/azure/aks/limit-egress-traffic is NOT mentioning EXPLICITLY the FQDN/port for Uptime SLA:
Is it management.azure.com/443 ???
Thanks

--

**Comment by kagowda** [comment]
Please follow the Required ports for AKS cluster section in the "limit-egress-traffic" docs. You need to consider all those ports for egress lockdown.

**Comment by brobichaud** [comment]
Whoa, I was surprised to see this. I had thought my cluster was already SLA backed. @jluk I wonder if Microsoft offers any insight into what the changes are on the Api server end when I opt into the SLA?

We primarily run Windows workloads on our AKS clusters and so I actually have two questions in this area:
1. The System nodepool is required even though we do not use it for anything. I'd like details on what this nodepool should look like (minimum SKU and nodecount). It should be documented in detail.
2. The Api server is a mysterious thing in AKS. It would be very helpful to understand some details about them, in particular now that there is this SLA backed concept what are the differences between them and non SLA backed?

**Comment by kagowda** [comment]
SLA provides a financially backed guarantee of 99.9% of API server availability with non-AZ clusters and 99.95% of API server availability for AZ enabled clusters. We provide these guarantees by allocating additional resources to meet the SLA. 
On a free cluster, we thrive to provide and Service level objective of 99.5%

**Comment by brobichaud** [comment]
> SLA provides a financially backed guarantee of 99.9% of API server availability with non-AZ clusters and 99.95% of API server availability for AZ enabled clusters. We provide these guarantees by allocating additional resources to meet the SLA.
> On a free cluster, we thrive to provide and Service level objective of 99.5%

Yeah, thanks, I did see those few details documented. But what does that really mean? For that $73/month is Microsoft running the Api servers on dedicated VMs vs shared VMs for no SLA? Are they merely adding redundancy? It does not add any color to the mystery that is the Api server in AKS. I guess it just feels weird to me to be paying for an SLA with no details. This is the very first Azure Resource where I have specifically needed to pay for it.

**Comment by seanmck** [comment]
We don't divulge implementation details of the control plane architecture because it is fully managed and subject to change. What we publish is what you should expect in terms of availability from the different options. You can then decide whether paying for that difference in expected result is worth it.

**Comment by dhananjaya-senanayake** [comment]
Any plan of providing Uptime SLA for existing AKS clusters ?

**Comment by flamingboo** [comment]
https://techcommunity.microsoft.com/t5/azure-kubernetes-service/aks-introduces-uptime-sla/ba-p/1350832
They are saying, it will be ... in coming months!

**Comment by kagowda** [comment]
Yes, we will release uptime SLA for existing clusters and expand the regions in the next 3 months.

**Comment by kagowda** [comment]
You can now enable SLA during cluster creation across all AKS regions. We are working to bring SLA support for existing clusters across all regions within a month's timeframe.

**Comment by kmoussa** [comment]
Hi @kagowda , can you please update the FAQ section here [https://azure.microsoft.com/en-us/pricing/details/kubernetes-service/ ](https://azure.microsoft.com/en-us/pricing/details/kubernetes-service/ )not to confuse ppl, This is what is mentioned 
![image](https://user-images.githubusercontent.com/12699056/84259193-60cf8900-ab18-11ea-919f-e2f47468e39c.png)

**Comment by kagowda** [comment]
Very happy to announce SLA for existing clusters in AKS public regions. We are still working to support Moon cake and Azure Government.

**Comment by dhananjaya-senanayake** [comment]
@kagowda , does this feature supports to be enabled from the Azure portal ?
